app/script/entity/message/Message.js

```diff
@@ -0,0 +1,351 @@
+    });
+
+    this.conversation_id = '';
+    this.from = '';
+    this.is_editing = ko.observable(false);
+    this.primary_key = undefined;
+    this.status = ko.observable(z.message.StatusType.UNSPECIFIED);
+    this.type = '';
+    this.user = ko.observable(new z.entity.User());
+    this.visible = ko.observable(true);
+    this.version = 1;
+
+    this.timestamp = ko.observable(Date.now());
+    this.should_effect_conversation_timestamp = true;
+
+    // z.message.MessageCategory
+    this.category = undefined;
+
+    this.display_timestamp_short = () => {
+      const date = moment.unix(this.timestamp() / 1000);
+      return date.local().format('HH:mm');
+    };
+
+    this.sender_name = ko.pureComputed(() => {
+      return z.util.get_first_name(this.user());
+    }
+      , this, {deferEvaluation: true});
+
+    this.accent_color = ko.pureComputed(() => {
+      return `accent-color-${this.user().accent_id()}`;
+    });
+  }
+
+  equals(message_et) {
+    return this.id === (message_et != null ? message_et.id : undefined);
+  }
+
+  /**
+   * Check if message contains an asset of type file.
+   * @returns {Boolean} Message contains any file type asset
+   */
+  has_asset() {
+    if (this.is_content()) {
+      for (const asset_et of Array.from(this.assets())) {
+        if (asset_et.type === z.assets.AssetType.FILE) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Check if message contains a file asset.
+   * @returns {Boolean} Message contains a file
+   */
+  has_asset_file() {
+    if (this.is_content()) {
+      for (const asset_et of Array.from(this.assets())) {
+        if (asset_et.is_file()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Check if message contains any image asset.
+   * @returns {Boolean} Message contains any image
+   */
+  has_asset_image() {
+    if (this.is_content()) {
+      for (const asset_et of Array.from(this.assets())) {
+        if (asset_et.is_image()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Check if message contains a location asset.
+   * @returns {Boolean} Message contains a location
+   */
+  has_asset_location() {
+    if (this.is_content()) {
+      for (const asset_et of Array.from(this.assets())) {
+        if (asset_et.is_location()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Check if message contains a text asset.
+   * @returns {Boolean} Message contains text
+   */
+  has_asset_text() {
+    if (this.is_content()) {
+      for (const asset_et of Array.from(this.assets())) {
+        if (asset_et.is_text()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Check if message contains a nonce.
+   * @returns {Boolean} Message contains a nonce
+   */
+  has_nonce() {
+    return [z.message.SuperType.CONTENT].includes(this.super_type);
```

Comment:
No longer needs an array check:

`return this.super_type === z.message.SuperType.CONTENT` or `return this.is_content()`